src/components/TemplateLibrary/index.js

```diff
@@ -177,8 +177,37 @@ const LibraryComponent = () => {
     const template = await Template.fromUrl(templateIndex.ciceroUrl);
     const ciceroMark = templateToCiceroMark(template);
     const templateIdentifier = template.getIdentifier();
-    setup(ciceroMark, template);
-    saveTemplateToXml(templateIdentifier);
+    Office.context.document.customXmlParts.getByNamespaceAsync(CUSTOM_XML_NAMESPACE, result => {
+      if (result.status === Office.AsyncResultStatus.Succeeded) {
+        if (result.value.length === 0) {
+          setup(ciceroMark, template);
+          saveTemplateToXml(templateIdentifier);
+        }
+        else {
+          const customXmlPart = result.value[0];
+          customXmlPart.getNodesAsync('*/*', result => {
+            if (result.status === Office.AsyncResultStatus.Succeeded) {
+              let f=1;
```

Comment:
Be more descriptive. Use good variable names such as "identifierExists".

Reply:
@algomaster99 
I will keep this in mind.